### PR TITLE
Add constructs for `Meshery Events`.

### PR DESCRIPTION
**Pull request by MUzairS15**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by MUzairS15** [comment]
https://github.com/meshery/meshkit/pull/35

File: models/events/build.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	operationId, _ := uuid.NewV4()
+	return &EventBuilder{
+		event: Event{
+			CreatedAt:   time.Now(),
+			OperationID: operationId,
+		},
+	}
+}
+
+func (e *EventBuilder) ActedOn(resource uuid.UUID) *EventBuilder {
```
Comment (leecalcote) [inline models/events/build.go:23]: ActedUpon, not "On"

File: models/events/build.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	e.event.Metadata = metadata
+	return e
+}
+
+func (e *EventBuilder) WithSeverity(severity EventSeverity) *EventBuilder {
+	e.event.Severity = severity
+	return e
+}
+
+func (e *EventBuilder) InitiatedBy(id uuid.UUID) *EventBuilder {
```
Comment (leecalcote) [inline models/events/build.go:48]: @MUzairS15 follow the table design spec exactly. Do not rename attributes.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: models/events/build.go (hunk 2 above)
Comment (MUzairS15) [reply models/events/build.go:48]: @leecalcote I have not diverged from the schema, the attribute name are exactly the same https://github.com/meshery/meshkit/pull/356/files/83dcd0952d500438cda2161e305b8ed1d12547a8#diff-5bc079c95a1811fd042d036db5a54fbae0ba3ef9e1645ce28d4eb4050b749049R49

File: models/events/build.go (hunk 1 above)
Comment (MUzairS15) [reply models/events/build.go:23]: https://github.com/meshery/meshkit/pull/356/files/83dcd0952d500438cda2161e305b8ed1d12547a8#diff-5bc079c95a1811fd042d036db5a54fbae0ba3ef9e1645ce28d4eb4050b749049R24

File: models/events/build.go (hunk 2 above)
Comment (leecalcote) [reply models/events/build.go:48]: Oh, I see. I'm confusing attribute name and function name. Symmetric function naming could be `fromUser` or InitiatingUser and InitiatingSystem.

File: models/events/build.go (hunk 1 above)
Comment (leecalcote) [reply models/events/build.go:23]: Why not use the same moniker for both function and attribute like what is done with Description?

You're not being consistent with the approach. In some instances you are adding a verb or preposition to the function name, but not always. Consistency facilitates legibility and comprehension by your future self and others.

**Review by leecalcote — APPROVED** [review]
The only thing missing now is the split of `event_types`

**Pull request merged**